### ISSUE TITLE
openPMD plugin: Crash with default parameters (dataPreparationStrategy = doubleBuffer) when compiling a hostonly Alpaka backend

### ISSUE THREAD

**Issue by franzpoeschel** [ComputationalRadiationPhysics/picongpu#3576]
I'm not sure if this should be considered a bug, but I ran a little simulation previously with `ALPAKA_ACC_CPU_B_SEQ_T_OMP2_ENABLE=ON`. Leaving the default parameters in the openPMD backend yields the error:
```
Unhandled exception of type 'St13runtime_error' with message 'DataConnector: Invalid dataset ID (MallocMCBuffer)', terminating
```
When specifying `--openPMD.dataPreparationStrategy mappedMemory`, the plugin runs fine, so the error is likely to stem from [this method](https://github.com/ComputationalRadiationPhysics/picongpu/blob/a36a5c5424a12ec321b3b8dbb3b77440304cb961/include/picongpu/plugins/openPMD/WriteSpecies.hpp#L128). Precise parameters found [here](https://github.com/openPMD/openPMD-example-datasets/pull/14#issuecomment-814692389).

Two questions:
1. Should the MallocMC thing work when picking a non-CUDA backend?
2. If the answer to (1) is "no", i.e. the double buffer strategy is a CUDA-specific thing, should we automatically use the `mappedMemory` strategy if other Alpaka backends are enabled?

**Comment by sbastrakov** [comment]
I am not sure how it's supposed to work, as not familiar with this part. But based on that function alone, perhaps [this](https://github.com/ComputationalRadiationPhysics/picongpu/blob/a36a5c5424a12ec321b3b8dbb3b77440304cb961/include/picongpu/plugins/openPMD/WriteSpecies.hpp#L133-L134) and [that](https://github.com/ComputationalRadiationPhysics/picongpu/blob/a36a5c5424a12ec321b3b8dbb3b77440304cb961/include/picongpu/plugins/openPMD/WriteSpecies.hpp#L170) pieces should be in the `#ifdef` guard (at start and end of that)?

**Comment by sbastrakov** [comment]
The error means that there was no object of type `MallocMCBuffer<DeviceHeap>` with the corresponding name registered. Upon looking in the code, we conditionally create and register it [here](https://github.com/ComputationalRadiationPhysics/picongpu/blob/a36a5c5424a12ec321b3b8dbb3b77440304cb961/include/picongpu/simulation/control/Simulation.hpp#L433-L434). So it would appear all usage of it should be inside similar `#ifdef` guards.

**Comment by franzpoeschel** [comment]
Thanks for the hints!

**Comment by psychocoderHPC** [comment]
fixed with #3580
